src/abstraction/nv.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    handles::{AuthHandle, NvIndexHandle, NvIndexTpmHandle, TpmHandle},
+    Context, Result,
+};
+
+// Source: TCG PC Client Platform TPM Profile Specification for TPM 2.0
+// Version 1.05, Revision 14, section 4.1, table 1
+const TPM_PT_NV_BUFFER_MAX: usize = 512;
```

Comment:
Two things worth mentioning here:
* there is already a (wrong) definition of this constant [here](https://github.com/parallaxsecond/rust-tss-esapi/blob/master/src/constants/tss.rs#L424) where it follows the Trusted Platform Module Library Part 2: Structures, page 51. I think the value in that `tss.rs` file is actually the property code that can be used with `GetCapability` to actually obtain the true value of `TPM_PT_NV_BUFFER_MAX` from the TPM. I'll raise a bug for this, maybe we should change those `PT`s to an enum to indicate what they're intended for.
* this 512 that you used is for PC clients only (as you stated in the comment). I think there's a need to either mark the function below as only working with PC TPMs or to make it more flexible by using `GetCapability`

Reply:
I'd been working on actually abstracting the Capability type, so I may as well just finish that and use that here.

Reply:
That would be great, thanks!

Reply:
That is now in PR #140, and I'll rebase this PR on top of that once that gets merged.

Reply:
I rebase on top of the in-progress #140 and added automatic detection of the max size.